src/DateInput.tsx

```diff
@@ -358,3 +360,79 @@ export function DateInputFieldWithLabel (
     </>
   )
 }
+
+// Type-in Date Input
+
+const dateMask = [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/]
+
+export function DateInputTypeIn ({
+  name,
+  hasError,
+  onChange
+}: {
+  name: string
+  hasError?: boolean
+  onChange(val: string): void
+}) {
+  return (
+    <MaskedInput
+      name={name}
+      mask={dateMask}
+      id='date'
+      placeholder='mm/dd/yyyy'
+      type='text'
+      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+        onChange(e.target.value)
+      }}
+      render={(inputRef, props) => (
+        <Input
+          name={name}
+          ref={inputRef}
+          hasError={hasError}
+          onChange={onChange}
+          {...props}
+        />
+      )}
+    />
+  )
+}
+
+export function DateInputTypeInField ({ name }: { name: string }) {
+  const validate = (value: string) => {
+    if (!value) {
+      return 'Required'
+    }
+
+    const moment = require('moment')
+    const month = parseInt(value.split('/')[0])
+    const day = parseInt(value.split('/')[1])
+    const year = parseInt(value.split('/')[2])
+
+    if (year < 1900) {
```

Comment:
I needed to prevent the user from entering years less than 4 digits (moment evaluates "199_" as year "0199" so that was passing validation).

We could exclude "_" from the string and check string length, but I figured this was more straightforward and we shouldn't expect any dates earlier than 1900.

I didn't set an upper bound in case we need future dates.

Reply:
Heh yeah that's probably fine. If we ever need to change it we can always add that functionality later.